lib/unicode_plot/braille_canvas.rb

```diff
@@ -36,17 +36,21 @@ def pixel!(pixel_x, pixel_y, color)
       pixel_x -= 1 unless pixel_x < pixel_width
       pixel_y -= 1 unless pixel_y < pixel_height
       tx = pixel_x.fdiv(pixel_width) * width
-      char_x = tx.floor + 1
-      char_x_off = pixel_x % X_PIXEL_PER_CHAR + 1
-      char_x += 1 if char_x < tx.round + 1 && char_x_off == 1
 
-      char_y = (pixel_y.fdiv(pixel_height) * height).floor + 1
-      char_y_off = pixel_y % Y_PIXEL_PER_CHAR + 1
+      char_x = tx.floor
```

Comment:
This code is both a fix for interpolation issue #32 but also removes several of the +1 / -1 manipulations needed which were there because the original UnicodePlots.jl (Julia) uses 1-based array indexing.